### PR TITLE
don't send AVATAR_SELF_ID over wire

### PR DESCRIPTION
**Pull request by sethalves**
- fix bug that made equipping fail when people are watching

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- rez the whiteboard from the market
- have a friend watch (also using this PR)
- use the markers from the whiteboard
- make sure it all looks right

**Comment by hifi-gustavo** [comment]
Build console output for commit: 37de4aefa6a31e892ccc5e461aa815de76d6f4f0 available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-37de4ae.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-37de4ae.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 174c674fd31195777fda03a631e11c4729c6e9e8 available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-174c674.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-174c674.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c7200d2d3a2571d5fe6894e72bf377d5554c1111 available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-c7200d2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-c7200d2.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 22f9fc70617280714bf9ed26fad3b0ad1ea90262 available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-22f9fc7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-22f9fc7.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 81bf5ee6554317c657c50df57dce1df4e73da77e available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-81bf5ee.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-81bf5ee.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3817230f1b734428cd606c5e7a8de28dcbc9e84 available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-e381723.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-e381723.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ab954412c13b88420f8c31259e627dc909e022ff available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-ab95441.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-ab95441.dmg

**Comment by claudioshg** [comment]
Will test with @Nex-Pro

File: libraries/shared/src/SpatiallyNestable.cpp [hunk 1]
```diff
@@ -1034,6 +1034,13 @@ AACube SpatiallyNestable::getQueryAACube() const {
 
 bool SpatiallyNestable::hasAncestorOfType(NestableType nestableType) const {
     bool success;
+    if (nestableType == NestableType::Avatar) {
+        QUuid parentID = getParentID();
+        if (parentID == AVATAR_SELF_ID) {
+            return true;
+        }
+    }
+
```
Comment (AndrewMeadows) [inline libraries/shared/src/SpatiallyNestable.cpp]: Non-blocking suggestion: move declaration of `success` above down below the early exit.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2ea07390c8f0572634ad12f362d9305f1d0a546e available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-2ea0739.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-2ea0739.dmg

**Comment by Nex-Pro** [comment]
I think this is a **FAIL**

In the course of testing we got several different fails.

One of which we could repro.

Take a marker.
Hold the marker above another user's hand with far grab or near.
Have second user  grip empty hand.
Drop marker on to  gripping hand of other avatar.
Observe marker appears very briefly on the arm of the receiving avatar then vanish in a rearward direction.

receiving avatar can see the marker. The avatar dropping it cannot.

Avatar with the marker sees marker and can draw. Observer who passed the marker can see drawing taking place but cannot see the marker.

Also applies to Eraser.

Claudio experienced being blasted across the domain by an erase he was holding.  Mayebe a mile or so.
He has additional notes.

Other fail is illustrated in the snapshot.

I have a JSON object of the marker with the attached line entities. This was triggered By me dropping a pen to my other hand. I was giving continuous haptic and the lines were being emmitted I guess up to the max number of line segments each time  basically until I dropped it.
Aslo had situation where , to me, Claudion was holding both an eraser and a hammer in the same hand. He could not see the eraser.
hifi-snap-by-piper-peppercorn-on-2017-02-06_21-59-49

hifi-snap-by-piper-peppercorn-on-2017-02-06_21-27-38

![hifi-snap-by-piper-peppercorn-on-2017-02-06_21-16-52]
(https://cloud.githubusercontent.com/assets/7314019/22667885/8dfa2d6e-ecb6-11e6-9f7f-5ea2814129fe.jpg)


hairy_hand_720
hifi-snap-by-claudioshg-on-2017-02-06_13-35-25_360

hiary_hand_2_720

**Comment by hifi-gustavo** [comment]
Build console output for commit: aaefc550f0e4ddcfda89242ef11528e7076017bf available at: https://gist.github.com/c64e2b4968b7d1f2654f1995df3f87ae<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-aaefc55.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-aaefc55.dmg

**Comment by claudioshg** [comment]
### Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested with Oculus Rift/Touch

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9609-2ea0739.exe

- The eraser has some quirks that the marker doesn't have when activating the disappearing bug. Whenever the eraser was invisible to @Nex-Pro, but visible to me, the eraser, unlike the marker, would launch me miles away from my original standing spot. This was reproduced many times. I held the eraser and then I would find myself in the Andromeda galaxy. :D As I was being flung at great distances, I noticed that once I let go of the eraser, I would stop flying away.

- At one point the eraser stopped my mobility. I was able to fly up and down, but not move forward or backward. Once I let go of the eraser and teleported, I was able to move again.

**Pull request merged**